### PR TITLE
[SYCL][Graph] Make debug print tests deterministic

### PR DESCRIPTION
**Pull request by mfrancepillois**
The order of root node in debug print tests was not deterministic. 
This PR updates these tests to make them deterministic by creating dependencies between nodes.

Addresses Issue: #12397

### CODE REVIEW

File: sycl/test-e2e/Graph/Inputs/debug_print_graph.cpp [hunk 1]
```diff
@@ -45,12 +45,12 @@ int main() {
       CGH.copy(AccB, AccA);
     });
 
-    add_node(Graph, Queue, [&](handler &CGH) {
-      auto AccA = BufferA2D.get_access<access::mode::read>(CGH);
```
Comment (EwanC) [inline sycl/test-e2e/Graph/Inputs/debug_print_graph.cpp:49]: Ca we remove `BufferA2D` all together from lines 30 & 31?

File: sycl/test-e2e/Graph/Inputs/debug_print_graph_verbose.cpp [hunk 2]
```diff
@@ -45,12 +45,12 @@ int main() {
       CGH.copy(AccB, AccA);
     });
 
-    add_node(Graph, Queue, [&](handler &CGH) {
-      auto AccA = BufferA2D.get_access<access::mode::read>(CGH);
```
Comment (EwanC) [inline sycl/test-e2e/Graph/Inputs/debug_print_graph_verbose.cpp:49]: Can we delete `BufferA2D` now?

**Review by EwanC — APPROVED** [review]

File: sycl/test-e2e/Graph/Inputs/debug_print_graph.cpp (hunk 1 above)
Comment (mfrancepillois) [reply sycl/test-e2e/Graph/Inputs/debug_print_graph.cpp:49]: I removed all unnecessary variables: BufferA2D among others.

File: sycl/test-e2e/Graph/Inputs/debug_print_graph_verbose.cpp (hunk 2 above)
Comment (mfrancepillois) [reply sycl/test-e2e/Graph/Inputs/debug_print_graph_verbose.cpp:49]: I removed all unnecessary variables: BufferA2D among others.

**Comment by EwanC** [comment]
tagging @intel/llvm-gatekeepers for merging

**Pull request merged**